### PR TITLE
WINC-814: Add GCP to machineset.sh

### PR DESCRIPTION
**Pull request by sebsoto**
Adds ability to spin up a Windows Server 2022 MachineSet on GCP.

### CODE REVIEW

File: hack/machineset.sh [hunk 1]
```diff
@@ -173,6 +173,56 @@ $(get_spec $infraID $az $provider)
...
+  fi
+
+  local infraID=$1
+  local region=$2
+  local az=$3
+  # For GCP the zone field returns the region + zone, like: `us-central1-a`.
+  # Installer created Machinesets only append the `-a` portion, so we should do the same.
+  local az_suffix=$(echo $az |awk -F "-" '{print $NF}')
+  local provider=$4
+  local projectID=$(cat ~/.gcp/osServiceAccount.json | jq -r '.project_id')
```
Comment (aravindhp) [inline hack/machineset.sh:189]: I suggest using `gcloud config get-value project`

File: hack/machineset.sh [hunk 2]
```diff
@@ -173,6 +173,56 @@ $(get_spec $infraID $az $provider)
...
+          projectID: ${projectID}
+          region: ${region}
+          serviceAccounts:
+          - email: ${infraID}-w@${projectID}.iam.gserviceaccount.com
+            scopes:
+            - https://www.googleapis.com/auth/cloud-platform
+          tags:
+          - ${infraID}-worker
+          userDataSecret:
+            name: worker-user-data
```
Comment (aravindhp) [inline hack/machineset.sh:221]: I know it doesn't matter, but lets use `windows-user-data` to be consistent.

File: hack/machineset.sh [hunk 3]
```diff
@@ -173,6 +173,56 @@ $(get_spec $infraID $az $provider)
...
+
+  local infraID=$1
+  local region=$2
+  local az=$3
+  # For GCP the zone field returns the region + zone, like: `us-central1-a`.
+  # Installer created Machinesets only append the `-a` portion, so we should do the same.
+  local az_suffix=$(echo $az |awk -F "-" '{print $NF}')
+  local provider=$4
+  local projectID=$(cat ~/.gcp/osServiceAccount.json | jq -r '.project_id')
+  local imageName=$(gcloud compute images list --filter="family = windows-2022-core" --sort-by="~creationTimestamp" --standard-images --limit=1 | tail -n1 |awk '{print $1}')
```
Comment (aravindhp) [inline hack/machineset.sh:190]: You can just do `gcloud compute images list --filter="family = windows-2022-core" --sort-by="~creationTimestamp" --standard-images --limit=1 --format="value(NAME)"`

**Review by aravindhp — CHANGES_REQUESTED** [review]
Thanks for doing this @sebsoto. Please address my comments.

File: hack/machineset.sh [hunk 4]
```diff
@@ -173,6 +173,56 @@ $(get_spec $infraID $az $provider)
 EOF
 }
 
+# get_gcp_ms creates a MachineSet for Google Cloud
```
Comment (jrvaldes) [inline hack/machineset.sh:176]: s/MachineSet/machineSet

everywhere.

File: hack/machineset.sh [hunk 5]
```diff
@@ -173,6 +173,56 @@ $(get_spec $infraID $az $provider)
...
+    error-exit incorrect parameter count for get_gcp_ms $#
+  fi
+
+  local infraID=$1
+  local region=$2
+  local az=$3
+  # For GCP the zone field returns the region + zone, like: `us-central1-a`.
+  # Installer created Machinesets only append the `-a` portion, so we should do the same.
+  local az_suffix=$(echo $az |awk -F "-" '{print $NF}')
+  local provider=$4
```
Comment (jrvaldes) [inline hack/machineset.sh:188]: Consider moving this right after line#184 close to the other parameters

**Review by saifshaikh48 — COMMENTED** [review]
LGTM

leaving the stamping to @jrvaldes

File: hack/machineset.sh [hunk 6]
```diff
@@ -257,6 +307,9 @@ case "$provider" in
     azure)
       ms=$(get_azure_ms $infraID $region $az $provider)
     ;;
+    gce)
+      ms=$(get_gcp_ms $infraID $region $az $provider)
```
Comment (jrvaldes) [inline hack/machineset.sh:311]: Given the provider is `gce`, does it makes sense to rename it as `get_gce_ms` ?

**Review by jrvaldes — COMMENTED** [review]
LGTM

@sebsoto  Thanks for working on this, minor suggestions PTAL

File: hack/machineset.sh (hunk 4 above)
Comment (sebsoto) [reply hack/machineset.sh]: Why? The object is `MachineSet`.

File: hack/machineset.sh (hunk 6 above)
Comment (sebsoto) [reply hack/machineset.sh:311]: even though the platform value is `gce` all documentation refers to the platform as GCP so i'd prefer to leave it as is

File: hack/machineset.sh (hunk 4 above)
Comment (aravindhp) [reply hack/machineset.sh]: +1 on using `MachineSet`

Comment (jrvaldes) [reply hack/machineset.sh]: Got it.

**Comment by sebsoto** [comment]
@aravindhp I've addressed your comments, PTAL when you get a chance

**Review by aravindhp — APPROVED** [review]

**Pull request merged**